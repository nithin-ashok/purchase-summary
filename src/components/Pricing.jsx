import React from "react";

const Pricing = ({subtotal, savings, tax, zip}) => {
    return (
        <table className="table">
            <tbody>
            <tr>
                <td>Subtotal</td>
                <td className="text-right font-bold">{`$${subtotal}`}</td>
            </tr>
            <tr>
                <td className="savings tooltip">Pickup savings
                    <span className="tooltiptext">
                       Picking up your order in the store cut costs, and we pass the savings on to you.
                   </span>
                </td>
                <td className="text-right font-bold font-red">{`-$${savings}`}</td>
            </tr>
            <tr>
                <td>Est. taxes & Fees
                    <span className="zip">(Based on {zip})</span>
                </td>
                <td className="text-right font-bold">{`$${tax}`}</td>
            </tr>
            </tbody>
        </table>
    );
}

export default Pricing;