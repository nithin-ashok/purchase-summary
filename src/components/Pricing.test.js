import React from 'react';
import Pricing from "./Pricing";
import renderer from 'react-test-renderer';

const pricing = {
    subtotal : 102.96,
    savings: 3.85,
    tax: 8.92,
    total: 108.03,
    zip: 85050
};


it("renderes correctly Pricing component", () => {
    const tree = renderer.
    create(<Pricing {...pricing}/>).toJSON();

    expect(tree).toMatchSnapshot();
});