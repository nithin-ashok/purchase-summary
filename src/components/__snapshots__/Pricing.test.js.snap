// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderes correctly Pricing component 1`] = `
<table
  className="table"
>
  <tbody>
    <tr>
      <td>
        Subtotal
      </td>
      <td
        className="text-right font-bold"
      >
        $102.96
      </td>
    </tr>
    <tr>
      <td
        className="savings tooltip"
      >
        Pickup savings
        <span
          className="tooltiptext"
        >
          Picking up your order in the store cut costs, and we pass the savings on to you.
        </span>
      </td>
      <td
        className="text-right font-bold font-red"
      >
        -$3.85
      </td>
    </tr>
    <tr>
      <td>
        Est. taxes & Fees
        <span
          className="zip"
        >
          (Based on 
          85050
          )
        </span>
      </td>
      <td
        className="text-right font-bold"
      >
        $8.92
      </td>
    </tr>
  </tbody>
</table>
`;
